alibuild_helpers/utilities.py

```diff
@@ -582,7 +584,20 @@ def getPackageList(packages, specs, configDir, preferSystem, noSystem,
             spec = replacement
             # Allows generalising the version based on the actual key provided
             spec["version"] = spec["version"].replace("%(key)s", key)
+            # Also replace %(key)s in environment variables
+            if "env" in spec:
+              for env_key, env_value in spec["env"].items():
+                if isinstance(env_value, str):
+                  spec["env"][env_key] = env_value.replace("%(key)s", key)
             recipe = replacement.get("recipe", "")
+            # Inject environment variables into the recipe since they're not available during execution
```

Comment:
This changes the current behaviour, no? env is not used for the current package, but only for it's dependencies.